pkg/ingester/ingester_v2.go

```diff
@@ -363,6 +413,15 @@ func (i *Ingester) v2Push(ctx context.Context, req *client.WriteRequest) (*clien
 				continue
 			}
 
+			if ve, ok := cause.(*validationError); ok {
```

Comment:
```suggestion
			var ve *validationError
			if errors.As(cause, &ve) {
```

This also works if errors are wrapped.